main.tf

```diff
@@ -80,3 +83,58 @@ data "aws_iam_policy_document" "cloudwatch-log-group" {
     resources = ["${aws_cloudwatch_log_group.this.arn}:*"]
   }
 }
+
+
+# Secret environment variables
+
+locals {
+  environment_variables = merge(var.environment_variables, local.secret_environment_variables)
+
+  secret_environment_variables = {
+    for k, v in var.secret_environment_variables : join("", [k, "_SECRET_ARN"]) => v
+  }
+}
+
+resource "aws_iam_role_policy" "secretsmanager-get-secret-value" {
+  count = length(var.secret_environment_variables) > 0 ? 1 : 0
+
+  name   = "secretsmanager-get-secret-value-${md5(data.aws_iam_policy_document.secretsmanager-get-secret-value[count.index].json)}"
+  role   = aws_iam_role.this.name
+  policy = data.aws_iam_policy_document.secretsmanager-get-secret-value[count.index].json
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# Whenever a change is made to the secrets, the role policy must be updated
+# first, then we need to wait for a few seconds and only then update the lambda function itself.
+# Waiting is necessary because otherwise the during initialization of the lambda function the
+# secrets-fetcher wrapper will not yet have the permissions for fetching the secret values.
```

Comment:
```suggestion
# Waiting is necessary because otherwise the during initialization of the lambda function
# it will not yet have the permissions for fetching the secret values.
```